Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/ActionsTreeView.cs

```diff
@@ -293,6 +288,20 @@ private void OnValidateCommand(ValidateCommandEvent evt)
             }
         }
 
+        private void OnPointerDown(PointerDownEvent evt)
+        {
+            // Allow right clicks to select an item before we bring up the matching context menu.
+            if (evt.button == (int)MouseButton.RightMouse && evt.clickCount == 1)
+            {
+                // Look upwards to the immediate child of the scroll view, so we know what Index to use
+                var element = (evt.target as VisualElement);
+                while (element.name != "unity-tree-view__item")
+                    element = element.parent;
+
+                m_ActionsTreeView.SetSelection(element.parent.IndexOf(element));
+            }
+        }
```

Comment:
I am a bit worried about setting selections like this. Is it even the expected behaviour to select things on right click? If then we should change it for actionmaps, actions and bindings to keep it consistent. Also I have the feeling there is a simpler way to do that then this, in particular looking at the while loop. (maybe there is even a way to allow right click selection in ActionsTreeView?) 
Also if the problem is just that the context menu doesn't show for the right clicked actionmap maybe the better way to fix this would be to hand over the correct input to the context menu?

Reply:
I am also suprised there is not a out-of-the-box solution for this in UITK that would provide consistent behavior across the editor. I would rather keep this behavior instead of introducing inconsistency with the rest of the editor. Could we reach out to UITK team as suggested by Rita and possible find similar examples in the editor?

Reply:
Some examples I found by quickly testing in the editor:
- Tags and layers in Project Settings list boxes (might be separate controls though), change focus/selection on right click
- Noticed that list in Package Manager settings do not select on right click but neither has a context menu either.
- Tree view in Project Settings (left panel) select on right click but do not have a context menu
- Project Hierarchy (probably most similar example), select on right click and has a context menu, so maybe UITK team can provide guidance on how that was achieved? Custom or some common mechanism? Also Game Object Hierarchy behaves this way

Reply:
Thanks for comments both!

> Is it even the expected behaviour to select things on right click?

I would think so - as well as the project/gameobject areas mentioned by Håkan, it's standard behaviour across operating systems, so I'd expect it to work here too.

> Also I have the feeling there is a simpler way to do that then this, in particular looking at the while loop.

Same! 😆 Maybe a question for #devs-uitoolkit if nobody has a suggestion? I'm still very new to it all.

Regarding asking UITK team about rightclick - the method here is one suggested by someone on their team just a few months ago, for the same issue: https://unity.slack.com/archives/C3414V4UV/p1693320293138709?thread_ts=1693306035.515169&cid=C3414V4UV and https://unity.slack.com/archives/C3414V4UV/p1694008889654529

Project and Hierarchy views are both IMGUI unfortunately, so can't make use of their methods 😞

Reply:
> I would think so - as well as the project/gameobject areas mentioned by Håkan, it's standard behaviour across operating systems, so I'd expect it to work here too.

Okay, wasn't aware of this, but that simplifies things!

I think #devs-uitoolkit is a good place to ask. Also both threads seem to be helpful, especially the trickle down recommendation for catching events before the context menu rightclick triggers.

Reply:
It think we should file a feature request for this. This might be a workaround now but it will for sure be needed  down the road for Project Hierarchy and then it would be nice to at least have a ticket to track/monitor.